find-files.nix

```diff
@@ -138,18 +139,27 @@ rec {
             else [];
 
         in
-          if isHighest
+          if isHighest || isForbiddenDir (parentDir)
           then
             {
               localIgnores = concatMap getIgnores dirs;
               worktreeRoot = p;
               inherit (dirInfo) gitDir;
             }
           else
-            go (dirOf p) dirs
+            go parentDir dirs
       ;
     in go dirPath [];
 
+  # isForbiddenDir: string -> bool
+  #
+  # Some directories should never be traversed when looking for .git
+  #  - for performance
+  #  - to help lorri and possibly other tools that monitor which paths are read
+  #    during evaluation
+  isForbiddenDir = p:
+    p == builtins.storePath || p == "/";
```

Comment:
```suggestion
    p == builtins.storeDir || p == "/";
```

Reply:
Good catch! My testing only required the `/` case.